provider/ec2/environ.go

```diff
@@ -1009,6 +1010,21 @@ func (e *environ) Subnets(instId instance.Id, subnetIds []network.Id) ([]network
 	return results, nil
 }
 
+func (e *environ) subnetsForVPC() (resp *ec2.SubnetsResp, vpcId string, err error) {
+	filter := ec2.NewFilter()
+	vpcId = e.ecfg().vpcID()
+	if !isVPCIDSet(vpcId) {
+		if hasDefaultVPC, err := e.hasDefaultVPC(); err == nil && hasDefaultVPC {
+			vpcId = e.defaultVPC.Id
+		}
+	}
+	if isVPCIDSet(vpcId) {
+		filter.Add("vpc-id", vpcId)
```

Comment:
we need to check what data we will get if we are 'juju bootstrap' into ec2-classic mode, and there exists VPCs in the region (that we can't use).